syrillian/src/engine/assets/cubemaps.rs

```diff
@@ -0,0 +1,388 @@
+use crate::engine::assets::{H, StoreTypeFallback};
+use crate::store_add_checked;
+use std::error::Error;
+use wgpu::{Device, Texture, TextureFormat, TextureView};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Cubemap {
+    pub faces: [Vec<u8>; 6],
+    pub width: u32,
+    pub height: u32,
+    pub format: TextureFormat,
+}
+
+pub type HCubemap = H<Cubemap>;
+
+impl H<Cubemap> {
+    const FALLBACK_CUBEMAP_ID: u32 = 0;
+    const MAX_BUILTIN_ID: u32 = 0;
+
+    pub const FALLBACK_CUBEMAP: H<Cubemap> = H::new(Self::FALLBACK_CUBEMAP_ID);
+}
+
+impl Cubemap {
+    const BYTES_PER_PIXEL: u32 = 4; // RGBA
+
+    const FACE_COLORS: [[u8; 4]; 6] = [
+        [255, 0, 0, 255],   // Right - Red
+        [0, 255, 0, 255],   // Left - Green
+        [0, 0, 255, 255],   // Top - Blue
+        [255, 255, 0, 255], // Bottom - Yellow
+        [255, 0, 255, 255], // Front - Magenta
+        [0, 255, 255, 255], // Back - Cyan
+    ];
+
+    pub fn gen_fallback_cubemap(size: u32) -> [Vec<u8>; 6] {
+        std::array::from_fn(|face_idx| {
+            Self::FACE_COLORS[face_idx].repeat(size as usize * size as usize)
+        })
+    }
+
+    /// Load cubemap from 6 individual face files
+    /// Order: [+X (right), -X (left), +Y (top), -Y (bottom), +Z (front), -Z (back)]
+    pub fn from_files(paths: [&str; 6]) -> Result<Self, Box<dyn Error>> {
+        // Load images from file paths
+        let mut faces = Vec::new();
+        let mut width = 0;
+        let mut height = 0;
+
+        for (i, path) in paths.iter().enumerate() {
+            // Try to load the image file
+            let img = match image::open(path) {
+                Ok(img) => img,
+                Err(_) => {
+                    // If file doesn't exist, use fallback color for that face
+                    let fallback_size = 64;
+                    let face_data = Self::FACE_COLORS[i].repeat(fallback_size * fallback_size);
+                    if i == 0 {
+                        width = fallback_size as u32;
+                        height = fallback_size as u32;
+                    }
+                    faces.push(face_data);
+                    continue;
+                }
+            };
+
+            // Convert to RGBA8
+            let rgba_img = img.to_rgba8();
+            if i == 0 {
+                width = rgba_img.width();
+                height = rgba_img.height();
+            }
+
+            // Ensure all faces have the same dimensions
+            let resized_img = if rgba_img.width() != width || rgba_img.height() != height {
+                image::imageops::resize(
+                    &rgba_img,
+                    width,
+                    height,
+                    image::imageops::FilterType::Lanczos3,
+                )
+            } else {
+                rgba_img
+            };
+
+            faces.push(resized_img.into_raw());
+        }
+
+        // Convert Vec to array
+        let faces_array: [Vec<u8>; 6] = faces
```

Comment:
If you push into this array immediately instead of into a Vec and then using try_into, this cannot fail..

Since you already take `paths: [&str; 6]` you should always have 6 entries anyways or abort.